### PR TITLE
Implemented `CBasePlayerWeapon::KickBack` hook 

### PR DESCRIPTION
**Pull request by ShadowsAdi**
And fixed compilation warning in `weapons.cpp` -> `GetBaseAccuracy()`

Related to https://github.com/s1lentq/reapi/issues/153
		https://dev-cs.ru/threads/1056/page-13#post-122698

### CODE REVIEW

**Comment by StevenKal** [comment]
New hooks in end of the list, always!

**Comment by ShadowsAdi** [comment]
Hm, you're right, gotta change that.

**Comment by StevenKal** [comment]
The following functions after it are broken, except if you recompile ReAPI module too with new IDs, but for people having the ReGameDLL_CS with "your kickback", and still having old ReAPI, and, using the following functions, this will be broken.

File: regamedll/dlls/API/CAPI_Impl.cpp [hunk 1]
```diff
@@ -205,6 +204,8 @@ GAMEHOOK_REGISTRY(CBasePlayer_Pain);
 GAMEHOOK_REGISTRY(CBasePlayer_DeathSound);
 GAMEHOOK_REGISTRY(CBasePlayer_JoiningThink);
 
+GAMEHOOK_REGISTRY(CBasePlayerWeapon_KickBack);
```
[the remark this answers is not in the document]
Comment (ShadowsAdi) [reply regamedll/dlls/API/CAPI_Impl.cpp:207]: I tested it both ways and seems that this way worked. It should've worked as it was before, but it didn't. More than this I can't say.

**Comment by s1lentq** [comment]
Done #849

**Pull request closed without merging**